### DISCUSSION TITLE
Registering Settings from a package ServiceProvider

### DISCUSSION THREAD

**Discussion by NeinMietz** [spatie/laravel-settings#140] [Q&A]
Hello,

how can a Setting be registered from a packages ServiceProvider? I need to register settings from my packages/modules automatically so the user should not have to modify any configs. Is this possible with this package?

**Comment by rubenvanassche** [comment]
At the moment we do not have such a feature since it could be that the package who wants to register settings is registered before the settings package. In such case your application would crash.

Registering settings within the boot method might be tricky aswel since they already could be required by other parts of the application.

The best seems to me to add a new class where we can statically add settings classes which need to be loaded. When Laravel has registered all classes, then the package can read the classes stored inside this static class and register them right before the boot process.

Maybe there are other solutions? I'm open for suggestions and PR's implement this feature 😄
